lib/logging.sh

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+
+#
+# Logging Library
+# Forked from
+# https://github.com/juan131/bash-libraries/blob/master/lib/liblog.bash
+
+# Color palette
+RESET='\033[0m'
+GREEN='\033[38;5;2m'
+RED='\033[38;5;1m'
+YELLOW='\033[38;5;3m'
+
+source config.sh
+# Log file location (default: "./logger.log")
+LOG_FILE_LOCATION=${LOG_FILE_LOCATION:-"./logger.log"}
+
+# trigger reviewdog
+# Functions
+# also should detect
+x=1
+# should detect
+y=2
+
+z=$y1
```

Comment:
⚠️ **[shellcheck]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>y1 is referenced but not assigned. [SC2154](https://github.com/koalaman/shellcheck/wiki/SC2154)
<!-- __reviewdog__:ChBiNzE5MzJiMjkzMTA1MDkyEgpzaGVsbGNoZWNr -->